splitting_data_by_type_year.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import os
+import re
+from datetime import datetime
+from typing import Dict, List
+
+import pandas as pd
+
+log_directory: str = r'./logs'
+os.makedirs(log_directory, exist_ok=True)
+logging.basicConfig(
+    filename=os.path.join(log_directory, 'data_split.log'),
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s',
+    encoding='utf-8'
+)
+
+
+def check_folder_exists(folder_path: str) -> None:
+    '''Check if folder exists'''
+    if not os.path.exists(folder_path):
+        logging.error(f'Folder {folder_path} does not exist')
+        raise FileNotFoundError(f'Folder {folder_path} does not exist')
+
+
+def get_csv_files(folder_path: str) -> List[str]:
+    '''Get list of CSV files in folder'''
+    files = os.listdir(folder_path)
+    csv_files = [os.path.join(folder_path, file)
+                 for file in files if file.endswith('.csv')]
+    return csv_files
+
+
+def split_files_by_category(csv_files: List[str]) -> Dict[str, List[str]]:
+    '''Split files into rent and sale categories'''
+    rent_files = [file for file in csv_files if 'rent' in file]
+    sale_files = [file for file in csv_files if 'rent' not in file]
```

Comment:
Read about [map(), filter(), reduce()](https://www.learnpython.org/en/Map%2C_Filter%2C_Reduce) methods. (Also[ Lambda func](https://www.w3schools.com/python/python_lambda.asp)) And rewrite this code using filter(). Or you can do this in one cycle:
```python
rent_files = []
sale_files = []

for element in csv_files:
    if 'rent' in file:
        rent_files.append(element)
    else:
        sale_files.append(element)

```

Reply:
+